core/services/ocr3/plugins/ccip/internal/reader/home_chain.go

```diff
@@ -0,0 +1,271 @@
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	libocrtypes "github.com/smartcontractkit/libocr/ragep2p/types"
+)
+
+type HomeChainPoller interface {
+	GetChainConfig(chainSelector cciptypes.ChainSelector) (ChainConfig, error)
+	GetAllChainConfigs() (map[cciptypes.ChainSelector]ChainConfig, error)
+	// GetSupportedChainsForPeer Gets all chain selectors that the peerID can read/write from/to
+	GetSupportedChainsForPeer(id libocrtypes.PeerID) (mapset.Set[cciptypes.ChainSelector], error)
+	// GetKnownCCIPChains Gets all chain selectors that are known to CCIP
+	GetKnownCCIPChains() (mapset.Set[cciptypes.ChainSelector], error)
+	// GetFChain Gets the FChain value for each chain
+	GetFChain() (map[cciptypes.ChainSelector]int, error)
+	services.Service
+}
+
+type HomeChainConfigPoller struct {
+	stopCh services.StopChan
+	services.StateMachine
+
+	homeChainReader types.ContractReader
+	// gets updated by the polling loop
+	chainConfigs map[cciptypes.ChainSelector]ChainConfig
+	// mapping between each node's peerID and the chains it supports. derived from chainConfigs
+	nodeSupportedChains map[libocrtypes.PeerID]mapset.Set[cciptypes.ChainSelector]
+	// set of chains that are known to CCIP, derived from chainConfigs
+	knownSourceChains mapset.Set[cciptypes.ChainSelector]
+	// map of chain to FChain value, derived from chainConfigs
+	fChain map[cciptypes.ChainSelector]int
+	lggr   logger.Logger
+	mutex  *sync.RWMutex
+	// How frequent will the poller fetch the chain configs
+	pollingInterval time.Duration
+}
+
+func NewHomeChainConfigPoller(
+	homeChainReader types.ContractReader,
+	lggr logger.Logger,
+	pollingInterval time.Duration,
+) *HomeChainConfigPoller {
+	return &HomeChainConfigPoller{
+		stopCh:              make(chan struct{}),
+		homeChainReader:     homeChainReader,
+		lggr:                lggr,
+		chainConfigs:        map[cciptypes.ChainSelector]ChainConfig{},
+		mutex:               &sync.RWMutex{},
+		nodeSupportedChains: map[libocrtypes.PeerID]mapset.Set[cciptypes.ChainSelector]{},
+		knownSourceChains:   mapset.NewSet[cciptypes.ChainSelector](),
+		fChain:              map[cciptypes.ChainSelector]int{},
+		pollingInterval:     pollingInterval,
+	}
+}
+
+func (r *HomeChainConfigPoller) Start(ctx context.Context) error {
+	err := r.fetchAndSetConfigs(ctx)
+	if err != nil {
+		// Just log, don't return error as we want to keep polling
+		r.lggr.Errorw("Initial fetch of on-chain configs failed", "err", err)
+	}
+	r.lggr.Infow("Start Polling ChainConfig")
+	return r.StartOnce(r.Name(), func() error {
+		go r.poll()
+		return nil
+	})
+}
+
+func (r *HomeChainConfigPoller) poll() {
+	ctx, cancel := r.stopCh.NewCtx()
+	defer cancel()
+	ticker := time.NewTicker(r.pollingInterval * time.Second)
```

Comment:
`pollingInterval` is already a `time.Duration`. Multiplying it again by a `time.Second` will not get you the results you seek; `time.Duration` is denominated in nanoseconds, so `time.Second` is actually `1e9` in raw integer terms, so if you specified `pollingInteval` as the logical 12 seconds (block time on ethereum mainnet) this ends up being `1e9 * 12e9 == 12e18` which is actually larger than the largest `int64`.

This is unless you are setting `pollingInterval` to be something like `1`, essentially a multiplier of the `time.Second` value, but this is not the correct usage of the `time.Duration` API. It'll just lead to confusion and bugs.

So this should just be 
```suggestion
	ticker := time.NewTicker(r.pollingInterval)
```

And the `pollingInterval` should be a meaningful `time.Duration` value (e.g `12 * time.Second`, maybe better as a constant instead of a struct field tbh)

Reply:
I started initially with `12 * time.Second`. @dimkouv had a comment to make it a variable. You don't see this as a changing parameter or something to experiment with in the future?